src/OpenTelemetry.Resources.Container/ContainerDetector.cs

```diff
@@ -131,45 +140,107 @@ private static string RemovePrefixAndSuffixIfNeeded(string input, int startIndex
         return input.Substring(startIndex, endIndex - startIndex);
     }
 
+    private string? ExtractK8sContainerId()
+    {
+        try
+        {
+            var baseUrl = this.k8sMetadataFetcher.GetServiceBaseUrl();
+            var containerName = this.k8sMetadataFetcher.GetContainerName();
+            if (string.IsNullOrEmpty(baseUrl) && string.IsNullOrEmpty(containerName))
```

Comment:
I think we agreed that we fail fast if either baseUrl or containerName is null.
So, `&&` should be replaced with `||`

Reply:
Fixed.